### PR TITLE
Convert nuimo app config format from .cfg to .yaml

### PR DESCRIPTION
**Pull request by commento**

### CODE REVIEW

**Comment by commento** [comment]
to be tested on the board

**Comment by commento** [comment]
note: the extension of nuimo_app file has to be changed also on production.ini

**Comment by commento** [comment]
tested on nano pi with nuimo controlling sonos

File: senic_hub/backend/tests/test_nuimo_components.py [hunk 1]
```diff
@@ -170,14 +170,15 @@ def test_put_invalid_component_returns_404(route_url, browser, temporary_nuimo_a
 def test_put_component_devices_modifies_app_config(component_url, browser, temporary_nuimo_app_config_file, settings):
     browser.put_json(component_url, {'device_ids': ['ph2-light-5', 'ph2-light-6']}, status=200)
 
-    config = ConfigParser()
-    config.read(settings['nuimo_app_config_path'])
-    component = dict(config['component-ph2'])
-    assert set(component['device_ids'].split(', ')) == set(['ph2-light-5', 'ph2-light-6'])
+    with open(settings['nuimo_app_config_path'], 'r+') as f:
```
Comment (larsblumberg) [inline senic_hub/backend/tests/test_nuimo_components.py]: You're only reading the file here – file mode `r` is sufficient, `r+` also opens the file for writing

File: senic_hub/backend/views/nuimo_components.py [hunk 2]
```diff
@@ -45,16 +45,16 @@ def nuimo_app_config_component_to_response_component(component_id, device):
...
         }
         return component
 
-    config = ConfigParser()
-    config.read(nuimo_app_config_path)
-    components = [
-        nuimo_app_config_component_to_response_component(s, dict(config[s]))
-        for s in config.sections()
-    ]
+    with open(nuimo_app_config_path, 'r+') as f:
```
Comment (larsblumberg) [inline senic_hub/backend/views/nuimo_components.py]: Use file open mode `r`

File: senic_hub/backend/views/nuimo_components.py [hunk 3]
```diff
@@ -45,16 +45,16 @@ def nuimo_app_config_component_to_response_component(component_id, device):
...
+        config = yaml.load(f)
+        components = [
```
Comment (larsblumberg) [inline senic_hub/backend/views/nuimo_components.py]: Assign components outside of `with` statement, it's not necessary to keep the file open here

File: senic_hub/backend/views/nuimo_components.py [hunk 4]
```diff
@@ -141,29 +141,27 @@ def create_component(device):
...
-        if k in ['ha_entity_id', 'id', 'ip_address', 'port', 'type', 'username']}
-    component['device_ids'] = app_config_component['device_ids'].split(', ')
+        if k in ['ha_entity_id', 'id', 'ip_address', 'port', 'type', 'username', 'device_ids']}
     return component
 
 
 @nuimo_component_service.get()
 def get_nuimo_component_view(request):
     component_id = request.matchdict['component_id']
     with open(request.registry.settings['nuimo_app_config_path'], 'r+') as f:
```
Comment (larsblumberg) [inline senic_hub/backend/views/nuimo_components.py]: file open mode `r` is enough

File: senic_hub/backend/views/nuimo_components.py [hunk 5]
```diff
@@ -141,29 +141,27 @@ def create_component(device):
...
-        config = ConfigParser()
-        config.read_file(f)
+        config = yaml.load(f)
         try:
```
Comment (larsblumberg) [inline senic_hub/backend/views/nuimo_components.py]: Please move try block outside of `with` statement

File: senic_hub/backend/commands.py [hunk 6]
```diff
@@ -46,8 +45,7 @@ def create_configuration_files_and_restart_apps(settings):
     # generate nuimo app config & restart supervisor app
     nuimo_app_config_file_path = settings['nuimo_app_config_path']
     with open(nuimo_app_config_file_path, 'w') as f:
-        config = generate_nuimo_app_configuration(devices)
-        config.write(f)
+        yaml.dump(generate_nuimo_app_configuration(devices), f, default_flow_style=False)
```
Comment (larsblumberg) [inline senic_hub/backend/commands.py]: I find easier to read if you first assign `generate_nuimo_app_configuration(devices)` to `config` as it was before and then pass `config` to `dump()`

File: senic_hub/backend/commands.py [hunk 7]
```diff
@@ -94,11 +92,12 @@ def phue_bridge_config(bridge):
...
-    config = configparser.ConfigParser()
+    config = None
     authenticated_devices = [d for d in devices if d["authenticated"]]
     for device in authenticated_devices:
         component = component_to_app_config_component(create_component(device))
-        config[component['id']] = component
+        if config is not None:
+            config[component['id']] = component
 
     return config
```
Comment (larsblumberg) [inline senic_hub/backend/commands.py]: Does this work? How I read it, the method always returns `None`

File: senic_hub/backend/tests/data/nuimo_app.yaml [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+    components:
+      ph2:
+        device_ids:
+          - ph2-light-4
+          - ph2-light-5
+          - ph2-light-6
+          - ph2-light-7
+          - ph2-light-8
+        ip_address: 127.0.0.2
+        type: philips_hue
```
Comment (larsblumberg) [inline senic_hub/backend/tests/data/nuimo_app.yaml]: `type` should be the first attribute as its the most important. Of course order doesn't matter, but it supports reading this file.

File: senic_hub/backend/tests/test_nuimo_components.py (hunk 1 above)
Comment (commento) [inline senic_hub/backend/tests/test_nuimo_components.py]: ok I will change it also for all the other tests

File: senic_hub/backend/commands.py (hunk 6 above)
Comment (commento) [inline senic_hub/backend/commands.py]: keeping the previous implementation result in this error:
```
        with open(nuimo_app_config_file_path, 'w') as f:
            config = generate_nuimo_app_configuration(devices)
>           config.write(f)
E           AttributeError: 'dict' object has no attribute 'write'

senic_hub/backend/commands.py:49: AttributeError
```

File: senic_hub/backend/views/nuimo_components.py (hunk 5 above)
Comment (larsblumberg) [inline senic_hub/backend/views/nuimo_components.py]: @commento can you please move try block outside of with statement

Comment (larsblumberg) [inline senic_hub/backend/views/nuimo_components.py]: I will take care

**Pull request merged**